packages/story-editor/src/app/api/apiProvider.js

```diff
@@ -71,75 +44,40 @@ function APIProvider({ children }) {
       storyLocking,
       pageTemplates: customPageTemplates,
     },
+    apiCallbacks,
     encodeMarkup,
     cdnURL,
   } = useConfig();
 
+  const {
+    getStoryById,
+    getStoryLockById,
+    setStoryLockById,
+    deleteStoryLockById,
+    getDemoStoryById,
+    saveStoryById,
+    autoSaveById,
+    getMedia,
+    uploadMedia,
+    updateMedia,
+    deleteMedia,
+    getLinkMetadata,
+    getAuthors,
+    getCurrentUser,
+    updateCurrentUser,
+    saveMetaBoxes,
+    getStatusCheck,
+    getCustomPageTemplates,
+    addPageTemplate,
+    deletePageTemplate,
+    getHotlinkInfo,
+  } = apiCallbacks;
+
   const pageTemplates = useRef({
     base: [],
     withoutImages: [],
   });
 
-  const getStoryById = useCallback(
-    (storyId) => {
-      const path = addQueryArgs(`${stories}${storyId}/`, {
-        context: 'edit',
-        _embed: STORY_EMBED,
-        web_stories_demo: false,
-        _fields: STORY_FIELDS,
-      });
-
-      return apiFetch({ path });
-    },
-    [stories]
-  );
-
-  const getStoryLockById = useCallback(
-    (storyId) => {
-      const path = addQueryArgs(`${stories}${storyId}/lock`, {
-        _embed: 'author',
-      });
-
-      return apiFetch({ path });
-    },
-    [stories]
-  );
-
-  const setStoryLockById = useCallback(
-    (storyId) => {
-      const path = `${stories}${storyId}/lock`;
-
-      return apiFetch({ path, method: 'POST' });
-    },
-    [stories]
-  );
-
-  const deleteStoryLockById = useCallback(
-    (storyId, nonce) => {
-      const data = new window.FormData();
-      data.append('_wpnonce', nonce);
-
-      const url = addQueryArgs(storyLocking, { _method: 'DELETE' });
-
-      window.navigator.sendBeacon?.(url, data);
-    },
-    [storyLocking]
-  );
-
-  const getDemoStoryById = useCallback(
-    (storyId) => {
-      const path = addQueryArgs(`${stories}${storyId}/`, {
-        context: 'edit',
-        _embed: STORY_EMBED,
-        web_stories_demo: true,
-        _fields: STORY_FIELDS,
-      });
-
-      return apiFetch({ path });
-    },
-    [stories]
-  );
-
   const getStorySaveData = useCallback(
```

Comment:
This function is very heavily linked to the WordPress implementation, like `base64Encode` and all the field mapping, why has this function remained here. This function feels like it should be moved.

Reply:
Done in https://github.com/google/web-stories-wp/pull/8966/commits/57e6c7e5626e2065dee39537c42f60881807558f